primary/src/metrics.rs

```diff
@@ -36,6 +42,42 @@ pub(crate) fn initialise_metrics(metrics_registry: &Registry) -> Metrics {
     }
 }
 
+#[derive(Default)]
+pub struct CoreCounters {
+    current_batches: AtomicU64,
+    last_commit_round: AtomicU64,
+    last_commit_batches: AtomicU64,
+}
+
+impl CoreCounters {
+    pub fn new() -> Arc<Self> {
+        Arc::new(Self::default())
+    }
+
+    pub fn set_commit_round(&self, round: u64) {
+        self.last_commit_round.store(round, Ordering::Relaxed);
+        let batches_now = self.current_batches.load(Ordering::Relaxed);
+        self.last_commit_batches
+            .store(batches_now, Ordering::Relaxed);
```

Comment:
> * It records the number of batches between the last commit (although not sure they are committed) and the current round (not sure they are included in the round).

ah I see now why. Agreed, I believe though we can come up with something more precise